Add ARM PL v23.04.1 for Ubuntu 22

dreambeyondorange opened the pull request:
ARM PL v23 changes the directory structure for the installed files to include the patch version.   
This requires refactoring the resource to include this change in the modulefile for armpl to identify the license

### Tests
* Ran `platform-install test arm-pl-*` for all OSes

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

gmarciani commented on CHANGELOG.md:
```diff
@@ -27,6 +27,7 @@ This file is used to list changes made in each version of the AWS ParallelCluste
   - Rdma-core: `rdma-core-46.0-1`
   - Open MPI: `openmpi40-aws-4.1.5-1`
 - Upgrade Slurm to version 23.02.3.
+- Upgrade ARM PL to version 23.04.1 for Ubuntu 22.04 only.
```
Why do we need to upgrade it only for Ubuntu22?

dreambeyondorange replied:
It’s only necessary for ubuntu 22 since the older version is not available for ubuntu 22

gmarciani reviewed (changes requested):
Spec tests failing

dreambeyondorange committed:
Add ARM PL v23.04.1 for Ubuntu 22

ARM PL v23 changes the directory structure for the installed files to include the patch version.
This requires refactoring the resource to include this change in the modulefile for armpl to identify the license
